documentation/src/main/asciidoc/user_guide/query.adoc

```diff
@@ -92,73 +144,468 @@ for (Object book : objectList) {
+in ram, filesystem, or Infinispan local caches.
+
+====== Effect of the index mode
+
+The `directory-based` and `near-real-time` index managers can be associated with different <<query.index-mode,index modes>>, resulting in different index distributions.
+
+REPL caches combined with the `ALL` index mode will result in a full copy of the cluster-wide index on each node. This mode allows queries to become effectively local without
+network latency. This is the recommended mode to index any REPL cache, and that's the choice picked by the <<query.autoconfig, auto-config>> when the a REPL cache is detected. The `ALL` mode should not be used with DIST caches.
+
+REPL or DIST caches combined with `LOCAL` index mode will cause each node to index only data inserted from the same JVM, causing an uneven distribution of the index. In order
+to obtain correct query results, it's necessary to use <<query.clustered-query-api,broadcast>> queries.
+
+REPL or DIST caches combined with `PRIMARY_OWNER` will also need broadcast queries. Differently from the `LOCAL` mode, each node's index will contain indexed entries which key is primarily owned by the node according to the consistent hash, leading to a more evenly distributed indexes among the nodes.
+
+
+====== directory-based index manager [[query.directory-based]]
+
+This is the default Index Manager used when no index manager is configured. The `directory-based` index manager is used to manage indexes backed by a local lucene directory. It supports _ram_, _filesystem_ and non-clustered
+_infinispan_ storage.
+
+
+[small]*Filesystem storage*
+
+This is the default storage, and used when index manager configuration is omitted. The index is stored in the filesystem using a link:https://lucene.apache.org/core/6_0_0/core/org/apache/lucene/store/MMapDirectory.html[MMapDirectory].
+It is the recommended storage for local indexes. Although indexes are persistent on disk, they get memory mapped by Lucene and thus offers decent query performance.
+
+Configuration:
+
+[source,xml]
+----
+<replicated-cache name="myCache">
+   <indexing index="ALL">
+      <!-- Optional: define base folder for indexes -->
+      <property name="default.indexBase">${java.io.tmpdir}/baseDir</property>
+   </indexing>
+</replicated-cache>
+----
+
+Infinispan will create a different folder under `default.indexBase` for each entity (index) present in the cache.
+
+[small]*Ram storage*
+
+Index is stored in memory using a link:https://lucene.apache.org/core/6_0_0/core/org/apache/lucene/store/RAMDirectory.html[Lucene RAMDirectory].
+Not recommended for large indexes or highly concurrent situations. Indexes stored in Ram are not persistent, so after a cluster shutdown a <<query.massindexer,re-index>>
+is needed. Configuration:
+
+
+[source,xml]
+----
+<replicated-cache name="myCache">
+   <indexing index="ALL">
+      <property name="default.directory_provider">ram</property>
+   </indexing>
+</replicated-cache>
+----
+
+[small]*Infinispan storage*
```

Comment:
It can be persistent by configuring those 3 caches with a cache store; even without a cache store, it's less "volatile" than `ram` since the cache will likely be replicated, so there's some redundancy in case of 1 server going down. That's a good point, I will add it to the doc